pkg/reconciler/service/resources/configuration.go

```diff
@@ -21,27 +21,47 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"knative.dev/pkg/kmeta"
+	cfgmap "knative.dev/serving/pkg/apis/config"
 	"knative.dev/serving/pkg/apis/serving"
 	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	labelerv2 "knative.dev/serving/pkg/reconciler/labeler/v2"
 	"knative.dev/serving/pkg/reconciler/service/resources/names"
 )
 
 // MakeConfiguration creates a Configuration from a Service object.
 func MakeConfiguration(service *v1.Service) (*v1.Configuration, error) {
+	return MakeConfigurationFromExisting(service, &v1.Configuration{}, cfgmap.Disabled)
+}
+
+// MakeConfigurationFromExisting creates a Configuration from a Service object given an existing Configuration.
+func MakeConfigurationFromExisting(service *v1.Service, existing *v1.Configuration, gc cfgmap.Flag) (*v1.Configuration, error) {
+	labels := map[string]string{serving.ServiceLabelKey: service.Name}
+	anns := kmeta.FilterMap(service.GetAnnotations(), func(key string) bool {
+		return key == corev1.LastAppliedConfigAnnotation
+	})
+
+	routeName := names.Route(service)
+	if gc != cfgmap.Enabled {
+		labels[serving.RouteLabelKey] = routeName
+	}
+
+	if gc == cfgmap.Enabled || gc == cfgmap.Allowed {
+		if set := labelerv2.GetListAnnValue(existing.Annotations, serving.RoutesAnnotationKey); set.Has(routeName) {
+			anns[serving.RoutesAnnotationKey] = existing.Annotations[serving.RoutesAnnotationKey]
+		} else {
+			anns[serving.RoutesAnnotationKey] = routeName
```

Comment:
Also here should we override or append to the list?

Reply:
we can do it that way. I suppose it would work if the same configuration were referenced by two services that way